src/main/java/mate/academy/springboot/datajpa/service/impl/ProductServiceImpl.java

```diff
@@ -0,0 +1,55 @@
+package mate.academy.springboot.datajpa.service.impl;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Map;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.repository.ProductRepository;
+import mate.academy.springboot.datajpa.repository.specification.SpecificationManager;
+import mate.academy.springboot.datajpa.service.ProductService;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ProductServiceImpl implements ProductService {
+    private final ProductRepository productRepository;
+    private final SpecificationManager<Product> productSpecificationManager;
+
+    public ProductServiceImpl(ProductRepository productRepository,
+                              SpecificationManager<Product> productSpecificationManager) {
+        this.productRepository = productRepository;
+        this.productSpecificationManager = productSpecificationManager;
+    }
+
+    @Override
+    public Product save(Product product) {
+        return productRepository.save(product);
+    }
+
+    @Override
+    public Product get(Long id) {
+        return productRepository.getById(id);
```

Comment:
dao layer should return optional for get by id methods

Reply:
What do you mean? We have repository that returns a model exactly